test/extensions/filters/http/cache/http_cache_implementation_test_common.cc

```diff
@@ -65,7 +65,13 @@ void HttpCacheImplementationTest::updateHeaders(
     absl::string_view request_path, const Http::TestResponseHeaderMapImpl& response_headers,
     const ResponseMetadata& metadata) {
   LookupContextPtr lookup_context = lookup(request_path);
-  cache()->updateHeaders(*lookup_context, response_headers, metadata);
+  auto update_promise = std::make_shared<std::promise<bool>>();
+  cache()->updateHeaders(*lookup_context, response_headers, metadata,
+                         [update_promise](bool result) { update_promise->set_value(result); });
```

Comment:
can we test the value of result after the promise is called?

Reply:
Made updateHeaders return the result, so it can be validated in either direction at the callsite, and added the corresponding validations (and also made the name of one of the tests more descriptive accordingly).